module/zfs/zil.c

```diff
@@ -628,7 +628,7 @@ zilog_dirty(zilog_t *zilog, uint64_t txg)
  * dirtied (zil_itx_assign) or cleaned (zil_clean) while we check its current
  * state.
  */
-boolean_t
+static boolean_t __attribute__((unused))
```

Comment:
I'm curious what the explanation is here?

Reply:
The unused annotations are for functions that are only used in debug assertions. Without the annotation, it triggers -Wunused-function if it's marked static and it's a non-debug build.

I could also fix that by ifdef-ing them out if DEBUG is not defined (or if NDEBUG is defined, whichever is preferable for such checks).

Reply:
Ok I might have guessed that. We have `__maybe_unused` that would help clarify this for future readers.

Reply:
Ah I didn't realize it's available on FreeBSD as well, I thought it would only work on linux is why I didn't use it, should have grep'ed first.

I'll change that then.